### ISSUE TITLE
Syncing resources multiple times creates duplicates.

### ISSUE THREAD

**Issue by vin-appdirect** [DataDog/datadog-sync-cli#203]
**Overview of issue:**

**Steps to reproduce the issue:**
1. Create dashboard in source org.
2. Use datadog-sync-cli to import diff and sync dashboard to destination org
3. dashboard is created in destination org.
4. Rerun the import diff and sync command and a duplicate dashboard is created in destination org.

**Describe the results you received:**
 Dashboard was duplicated in destination org when the sync command is rerun.

**Describe the results you expected:**
Dashboard should not be duplicated in destination org when the sync command is rerun

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by therve** [comment]
Hi,

Can you describe how you run the tool? State is supposed to be saved locally, this is what should prevent duplicates. What's the version you're using?

Thanks.

**Comment by vin-appdirect** [comment]
Hi,

I am using a config file to store the source and destination app and api keys. and mentioning resources and filter in the same config file.  Thats how the config file looks like (didnt mention keys here though).

`resources="dashboards"`
`filter=["Type=Dashboards;Name=title;Value=hello-world"]`

Then we run the below commands.

`datadog-sync import --config config.properties`
`datadog-sync diffs --config config.properties`
`datadog-sync sync --config config.properties`

**Comment by therve** [comment]
Can you verify what's present in the `resources` directory? Do you keep its content between runs?

**Comment by vin-appdirect** [comment]
I tihnk that was the reason. The resources directory was removed and as a result it created the resources with a new ID and synced those. This can be closed. thanks

**Comment by chino** [comment]
I had something similar happen where they got duplicated. Are we saying here we have to store the resources folder forever? Does it not simply check the target list before adding it again? Is there no unique ID such as the title which can be the key?

**Comment by therve** [comment]
Yes you have to keep the resources folder around. Things like titles are not unique identifier, so we need to keep a mapping of IDs between the 2 orgs.

**Comment by chino** [comment]
In any case is there no way that refreshing the resources can pull a unique id from the server side? Having to keep a cache folder around isn't practical.

If you send to multiple sources\destinations does the single resources folder remember them all?

**Comment by therve** [comment]
> In any case is there no way that refreshing the resources can pull a unique id from the server side? Having to keep a cache folder around isn't practical.

As of today there is no such mechanism no.

> If you send to multiple sources\destinations does the single resources folder remember them all?

No if you have multiple source and destination organizations you should keep different resources folder.
